source/grid/manifold_lib.cc

```diff
@@ -2100,15 +2109,20 @@ TransfiniteInterpolationManifold<dim, spacedim>::pull_back(
       for (unsigned int d = 0; d < spacedim; ++d)
         for (unsigned int e = 0; e < dim; ++e)
           Jinv_deltaf[e] += inv_grad[d][e] * delta_f[d];
-      const Tensor<1, dim> factor =
-        (delta_x - Jinv_deltaf) / (delta_x * Jinv_deltaf);
-      Tensor<1, spacedim> jac_update;
-      for (unsigned int d = 0; d < spacedim; ++d)
-        for (unsigned int e = 0; e < dim; ++e)
-          jac_update[d] += delta_x[e] * inv_grad[d][e];
-      for (unsigned int d = 0; d < spacedim; ++d)
-        for (unsigned int e = 0; e < dim; ++e)
-          inv_grad[d][e] += factor[e] * jac_update[d];
+
+      // prevent division by zero
+      if (delta_x * Jinv_deltaf != 0)
+        {
+          const Tensor<1, dim> factor =
```

Comment:
unrelated to this PR, but would you mind adding a comment with the formula here?